### ISSUE TITLE
Repo Analytics

### ISSUE THREAD

**Issue by DanceParty** [nicholasbergesen/octo-analytics#4]
## Description

**TLDR:** We want to be able to query for all workflow failures by event type on a Repository

The goal here is to be able to query all workflow failures (before the first success) per event in the workflow. Using this, we can compare failure rates between the different workflow events to see which events may be harder to use or have lower failure rates.

A potential next step would be to figure out the total failures of an event on the repository compared to the non-consecutive successes. This way we can have a little more information on event's failures vs successes (i.e. push will fail an average of 5 times for every 1 success). Important note here is not tracking consecutive successes as we think this would inflate the numbers for no changes to the event on the workflow file.

Ex of the counting (Without the potential next step):

Workflow 1: F - F - F - S - F - S
Workflow 2: F - F - S - S - F - S

Total failures: 4 (before the first success)

Ex of the counting (with the next step factored in)

Workflow 1: F - F - F - S - F - S
Workflow 2: F - F - S - S - F - S

Total Failure: 7
Total Success: 4 (5 successes - 1 consecutive success)
The failure rate would be 4 / 11
